### ISSUE TITLE
netjack crash issue in JackMidiPort.cpp

### ISSUE THREAD

**Issue by FlorianBador** [jackaudio/jack2#151]
Hi there,

All machines are:
jack-1.9.10
Linux 3.14.12-rt9
Slackware64 based, most things recompiled native.

I run netjack at 96KHz with: jack_netsource -H 192.168.200.2 -i64 -o0 -O24 -I0 -b32 -n1 -p 5000 -N sampler
This is to connect Ardour on a machine to LinuxSampler on another.
Everything works fine most of the time without xruns, but from time to time I get this crash, especially when under pressure of many midi notes:

jack_netsource: ../common/JackMidiPort.cpp:133: void Jack::MidiBufferMixdown(void, void*, int, jack_nframes_t): Assertion `next_event != 0' failed.

It is always the same and happens once or twice an hour in my workflow.
I can't find anything related to that.
Any idea where to start / what I can do?

The source line 133: https://github.com/jackaudio/jack2/blob/master/common/JackMidiPort.cpp#L133
Seems like a similar issue but gives no clue: https://code.google.com/p/amsynth/issues/detail?id=64

Thanks a lot!

**Comment by FlorianBador** [comment]
PS: please note that the more "abused" the machine is in cpu usage, the more often it happens. e.g. I am running some cryptography programs right now in the same time (although niced a lot) and it happens way more frequently.

**Comment by falkTX** [comment]
perhaps that issue as #46

**Comment by sletz** [comment]
I should be fixed on git version (1.9.11) : can you try it?


Le 15 oct. 2015 � 17:22, Florian Bador <notifications@github.com> a �crit :

**Comment by FlorianBador** [comment]
Current git shows 1.9.10-147-g38f946f

**Comment by FlorianBador** [comment]
FIXED.
Unbelievable...
Not only that I can no longer reproduce but I stress-tested it for about an hour 1/2 with a colossal amount of MIDI & audio data and I could not even get an xrun! :)
 
TEST DETAILS:
Ardour had 34 MIDI and 170 audio tracks, was sending 24 MIDI via netjack and receiving back 64 audio from Linux Sampler. The whole thing at 96KHz 32 bit float with just 5ms of latency. I usually have a very large amount of MIDI data (orchestra) but I multiplied everything by about 15 so I ended up having a amount of data that no music can ever reach.
On the top of that, both machines were also running some multi-threaded cpu-heavy cryptography  programs and these machines are affordable.
 
That's incredible... I think there was probably some serious optimizations in jack since the official release of 1.19.10 also, but I'm very curious to see what you've changed exactly? (I'm not the only one, we talked about it on the IRC)
 
Thanks again!
